src/Asset/Asset.php

```diff
@@ -0,0 +1,393 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2023 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Asset;
+
+use CommonDBTM;
+use Glpi\Application\View\TemplateRenderer;
+use Entity;
+use Group;
+use Location;
+use Manufacturer;
+use State;
+use Toolbox;
+use User;
+
+abstract class Asset extends CommonDBTM
+{
+    /**
+     * Search option ID used for the asset definition system criteria.
+     */
+    private const DEFINITION_SO_ID = 250;
+
+    final public function __construct()
+    {
+        foreach (static::getDefinition()->getEnabledCapacities() as $capacity) {
+            $capacity->onObjectInstanciation($this);
+        }
+    }
+
+    /**
+     * Get the asset definition related to concrete class.
+     *
+     * @return AssetDefinition
+     */
+    public static function getDefinition(): AssetDefinition
+    {
+        if (!property_exists(static::class, 'definition')) {
+            throw new \RuntimeException('Asset definition is expected to de defined in concrete class.');
+        }
+
+        return static::$definition ?? null;
```

Comment:
I did not find a good way to do this.

Declaring an `abstract public static function getDefinition(): AssetDefinition` method could be a solution, but `atoum` is not able to handle it. Indeed, it breaks tests declared in `tests/functional/Glpi/Asset/Asset.php` as it will automatically tries to create a mock class (that we do not even ask for) and fails with the following error:
```
Error FATAL ERROR in /var/www/glpi/tests/functional/Glpi/Asset/Asset.php on unknown line, generated by file /var/www/glpi/vendor/atoum/atoum/classes/mock/generator.php(219) : eval()'d code on line 2:
Class mock\Glpi\Asset\Asset contains 1 abstract method and must therefore be declared abstract or implement the remaining methods (Glpi\Asset\Asset::getDefinition)
```

The `?? null` was here to fix the `Access to an undefined static property static(Glpi\Asset\Asset)::$definition.` PHPStan issue.

I changed the way it is done in 8a69821c006fae4be60e78e9ad9ea555faddcd06.

If you find a better way to handle this, feel free to propose a patch.

Anyway, as the only expected place to extend this abstract class is in `AssetDefinitionManager::loadConcreteClass()`, this unconventional way of doing things should not cause too many issues.